pkg/sync/syncer.go

```diff
@@ -60,6 +61,8 @@ type ProgressCounts struct {
 	GrantsProgress       map[string]int
 	LastGrantLog         map[string]time.Time
 	LastActionLog        time.Time
+	mu                   sync.RWMutex // Protect concurrent access to maps
```

Comment:
The TODO on line 70 can be removed. :)

Also maybe it makes sense to put the progress stuff in progress.go.